src/associations/belongs-to-many.ts

```diff
@@ -0,0 +1,1316 @@
+      };
+    }
+
+    if (this.scope) {
+      Object.assign(values, this.scope);
+      if (options.fields) {
+        options.fields = [...options.fields, ...Object.keys(this.scope)];
+      }
+    }
+
+    // Create the related model instance
+    const newAssociatedObject = await this.target.create(values, options);
+
+    await this.add(sourceInstance, newAssociatedObject, omit(options, ['fields']));
+
+    return newAssociatedObject;
+  }
+}
+
+// workaround https://github.com/evanw/esbuild/issues/1260
+Object.defineProperty(BelongsToMany, 'name', {
+  value: 'BelongsToMany',
+});
+
+export function isThroughOptions<M extends Model>(val: any): val is ThroughOptions<M> {
+  return isPlainObject(val) && 'model' in val;
+}
+
+function normalizeThroughOptions<M extends Model>(
+  source: ModelStatic<any>,
+  target: ModelStatic<any>,
+  through: ThroughOptions<M>,
+  sequelize: Sequelize,
+): NormalizedThroughOptions<M> {
+  const timestamps = through.timestamps ?? sequelize.options.define?.timestamps;
+
+  let model;
+
+  if (!through || (typeof through.model !== 'string' && !isModelStatic(through.model))) {
+    throw new AssociationError(`${source.name}.belongsToMany(${target.name}) requires a through model, set the "through", or "through.model" options to either a string or a model`);
+  }
+
+  if (isModelStatic<M>(through.model)) {
+    model = through.model;
+  } else if (sequelize.isDefined(through.model)) {
+    model = sequelize.model(through.model) as ModelStatic<M>;
+  } else {
+    model = sequelize.define(through.model, {} as ModelAttributes<M>, {
+      tableName: through.model,
+      indexes: [], // we don't want indexes here (as referenced in #2416)
+      paranoid: through.paranoid || false, // Default to non-paranoid join (referenced in #11991)
+      validate: {}, // Don't propagate model-level validations
+      timestamps: through.timestamps,
+
+      // @ts-expect-error -- TODO: make 'schema' a public property on Model once the method has been removed (sequelize 8)
+      schema: source._schema,
+      // @ts-expect-error
+      schemaDelimiter: source._schemaDelimiter,
+    });
+  }
+
+  return {
+    ...through,
+    timestamps,
+    model,
+  };
+}
+
+function normalizeOptions<SourceKey extends string, TargetKey extends string, ThroughModel extends Model>(
+  type: AssociationStatic<any>,
+  options: BelongsToManyOptions<SourceKey, TargetKey, ThroughModel>,
+  source: ModelStatic<Model>,
+  target: ModelStatic<Model>,
+): NormalizedBelongsToManyOptions<SourceKey, TargetKey, ThroughModel> {
+
+  if ('timestamps' in options) {
+    throw new TypeError('The "timestamps" option in belongsToMany has been renamed to through.timestamps');
+  }
+
+  if ('uniqueKey' in options) {
+    throw new TypeError('The "uniqueKey" option in belongsToMany has been renamed to through.unique');
+  }
+
+  const sequelize = target.sequelize!;
+
+  // TODO: normalize otherKey using the same logic as in normalizeBaseAssociationOptions
```

Comment:
When should we take care of this?

Reply:
Oh, it's done. That's an old TODO. Removed